testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_delta_sync.py

```diff
@@ -94,9 +99,24 @@ def test_delta_sync_replication(params_from_base_test_setup, num_of_docs, replic
         cbl_db_docs = db.getDocuments(cbl_db, doc_ids)
         for doc_id, doc_body in cbl_db_docs.items():
             for i in range(number_of_updates):
-                doc_body["new_field_1"] = random_string(length=30)
-                doc_body["new_field_2"] = random_string(length=80)
-                db.updateDocument(database=cbl_db, data=doc_body, doc_id=doc_id, attachments_name=file_attachment)
+                if file_attachment:
+                    mutable_dictionary = dictionary.toMutableDictionary(doc_body)
+                    dictionary.setString(mutable_dictionary, "new_field_1", random_string(length=30))
```

Comment:
all set methods return Mutable Dictionary object, however, you are nowhere retaining that object. Does object from toMutableDictionary call retrains all future changes?

This is for all set methods?

Reply:
When I set values, it effects doc_body, I do not require mutable dictionary . For update document, I need doc_body. Also compare docs assertion will make sure new fields and attachments are replicated which proves that new changes are effected.